File: lib/domain/repositories/registered_library_repository.dart
import 'package:libcheck/domain/models/library.dart';

abstract class RegisteredLibraryRepository {
  Future<List<Library>> getAll();
  Future<void> saveAll(List<Library> libraries);
  Future<List<Library>> add(Library library);
  Future<List<Library>> addAll(List<Library> libraries);
  Future<List<Library>> remove(Library library);
}

File: src/domain/repositories/registeredLibraryRepository.ts
import { Library } from '@/domain/models/library';

export interface RegisteredLibraryRepository {
  getAll(): Promise<Library[]>;
  saveAll(libraries: Library[]): Promise<void>;
  add(library: Library): Promise<Library[]>;
  addAll(libraries: Library[]): Promise<Library[]>;
  remove(library: Library): Promise<Library[]>;
}
